spec/tools/environment_builders/openstack_environments.rb

```diff
@@ -108,6 +121,98 @@ def deactivate_paginations
   end
 end
 
+def change_variable_atributes(vcr_env_file_name)
+  # Set all dates, tokens, etc. to one value, so we don't have diff in every rebuilt VCR and therefore git conflict.
+  # Without this, each 2 VCRs would have git conflict, even when no API request changed,
+  vcr_env = YAML.load_file(vcr_env_file_name)
+  vcr_env["http_interactions"].each do |request|
+    request["request"]["headers"]["User-Agent"] = "fog/2.0.0.pre.0"
+    request["request"]["headers"]["X-Auth-Token"] = OBFUSCATED_X_AUTH_TOKEN
+    request["response"]["headers"]["Date"] = OBFUSCATED_DATE
+    unless (response_header = request.fetch_path("response", "headers")).blank?
+      response_header["X-Compute-Request-Id"] = OBFUSCATED_REQUEST_ID if response_header["X-Compute-Request-Id"]
+      response_header["X-Openstack-Request-Id"] = OBFUSCATED_REQUEST_ID if response_header["X-Openstack-Request-Id"]
+      response_header["X-Timestamp"] = OBFUSCATED_TIMESTAMP if response_header["X-Timestamp"]
+      response_header["X-Put-Timestamp"] = OBFUSCATED_TIMESTAMP if response_header["X-Put-Timestamp"]
+      response_header["X-Trans-Id"] = OBFUSCATED_TRANS_ID if response_header["X-Trans-Id"]
+      response_header["X-Subject-Token"] = OBFUSCATED_X_AUTH_TOKEN if response_header["X-Subject-Token"]
+    end
+
+    request["recorded_at"] = OBFUSCATED_DATE
+    request["response"]["http_version"] = ""
+
+    proces_request_body!(request)
+    proces_response_body!(request)
+  end
+  File.open(vcr_env_file_name, 'w') { |f| f.write vcr_env.to_yaml }
+end
+
+def proces_request_body!(request)
+  begin
+    body = JSON.parse(request["request"]["body"]["string"])
+  rescue
+    # Skip no valid JSONs, e.g when 404 is returned
+    return
+  end
+  return if body.blank? || !body.kind_of?(Hash)
+  body["auth"]["token"]["id"] = OBFUSCATED_X_AUTH_TOKEN if body.fetch_path("auth","token","id")
```

Comment:
there is also store_path, in case you weren't aware.

```ruby
body.store_path("auth", "token", "id", OBFUSCATED_X_AUTH_TOKEN) if body.fetch_path("auth", "token", "id")
```

then you could move the path into a variable if you chose.

```ruby
path = %w(auth token id)
body.store_path(path, OBFUSCATED_X_AUTH_TOKEN) if body.fetch_path(path)
```